### PR TITLE
feat(alerts): show a warning about implicit alerts deletion

### PR DESCRIPTION
**Pull request by nikitaevg**
## Problem

https://github.com/PostHog/posthog/issues/14331

## Changes

This adds a warning banner to warn about implicit deletion of alerts.

[Screencast](https://github.com/PostHog/posthog/assets/6707908/ced99999-f284-49e0-9ca1-089e9229599b)

This also removes alerts when the updated insight doesn't support alerts.

[Screencast](https://github.com/PostHog/posthog/assets/6707908/daa3cce1-c638-4529-8011-f62ccdec34e4)

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Manually + automated tests

### CODE REVIEW

**Comment by nikitaevg** [comment]
@mariusandra PTAL!

File: posthog/models/alert.py [hunk 1]
```diff
@@ -1,4 +1,21 @@
 from django.db import models
+from posthog.schema import ChartDisplayType, TrendsQuery
+from sentry_sdk import capture_exception
+from posthog.hogql_queries.legacy_compatibility.filter_to_query import filter_to_query
+from posthog.models.insight import Insight
+
+
+def are_alerts_supported_for_insight(insight: Insight) -> bool:
+    try:
+        query = filter_to_query(insight.filters)
```
Comment (mariusandra) [inline posthog/models/alert.py:10]: Some insights already only have `query` set, and no `filters`.

**Review by webjunkie — CHANGES_REQUESTED** [review]

File: frontend/src/lib/components/Alerts/alertDeletionWarningLogic.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { actions, connect, kea, listeners, path, props, reducers } from 'kea'
+import { insightVizDataLogic } from 'scenes/insights/insightVizDataLogic'
+
+import { InsightLogicProps, InsightShortId } from '~/types'
+
+import type { alertDeletionWarningLogicType } from './alertDeletionWarningLogicType'
+import { alertsLogic, areAlertsSupportedForInsight } from './alertsLogic'
+
+export const alertDeletionWarningLogic = kea<alertDeletionWarningLogicType>([
```
[the remark this answers is not in the document]
Comment (nikitaevg) [reply frontend/src/lib/components/Alerts/alertDeletionWarningLogic.ts]: Yeah, makes sense, done

**Review by webjunkie — APPROVED** [review]

**Comment by nikitaevg** [comment]
@webjunkie Could you please merge this?

**Pull request merged**